zqd/handlers_zeek_test.go

```diff
@@ -11,88 +12,142 @@ import (
+type packetPostTest struct {
+	Name           string
+	ZeekExec       string
+	Pcapfile       string
+	ExpectedStatus int
+	Test           func(packetPostResult)
 }
 
-func (s *PacketPostSuite) TestCount() {
-	expected := `
+func TestZeekPacketPost(t *testing.T) {
+	var cases = []packetPostTest{
+		{
+			Name: "SearchCount",
+			Test: func(p packetPostResult) {
+				expected := `
 #0:record[count:uint64]
 0:[4;]`
-	res := execSearch(s.T(), s.core, s.space, "count()")
-	s.Equal(test.Trim(expected), res)
-}
-
-func (s *PacketPostSuite) TestNoPacketFilter() {
-	expected := ""
-	res := execSearch(s.T(), s.core, s.space, "_path = packet_filter | count()")
-	s.Equal(expected, res)
-}
-
-func (s *PacketPostSuite) TestSpaceInfo() {
-	u := fmt.Sprintf("http://localhost:9867/space/%s", s.space)
-	res := httpSuccess(s.T(), zqd.NewHandler(s.core), "GET", u, nil)
-	var info api.SpaceInfo
-	err := json.NewDecoder(res).Decode(&info)
-	s.NoError(err)
-	s.Equal(s.pcapfile, info.PacketPath)
-	s.True(info.PacketSupport)
-	s.Equal(nano.Ts(1501770877471635000), *info.MinTime)
-	s.Equal(nano.Ts(1501770880988247000), *info.MaxTime)
-}
-
-func (s *PacketPostSuite) TestWritesIndexFile() {
-	stat, err := os.Stat(filepath.Join(s.core.Root, s.space, packet.IndexFile))
-	s.NoError(err)
-	s.NotNil(stat)
+				res := execSearch(p.t, p.core, p.space, "count()")
+				p.Equal(test.Trim(expected), res)
+			},
+		},
+		{
+			Name: "SearchTimelessPoints",
+			Test: func(p packetPostResult) {
+				expected := ""
+				res := execSearch(p.t, p.core, p.space, "ts = 0 | count()")
+				p.Equal(expected, res)
+			},
+		},
+		{
+			Name: "SpaceInfo",
+			Test: func(p packetPostResult) {
+				u := fmt.Sprintf("http://localhost:9867/space/%s", p.space)
+				res := httpSuccess(p.t, zqd.NewHandler(p.core), "GET", u, nil)
+				var info api.SpaceInfo
+				err := json.NewDecoder(res).Decode(&info)
+				p.NoError(err)
+				p.Equal(p.pcapfile, info.PacketPath)
+				p.True(info.PacketSupport)
+			},
+		},
+		{
+			Name: "PacketIndexExists",
+			Test: func(p packetPostResult) {
+				stat, err := os.Stat(filepath.Join(p.core.Root, p.space, packet.IndexFile))
+				p.NoError(err)
+				p.NotNil(stat)
+			},
+		},
+		{
+			Name: "StatusMessage",
+			Test: func(s packetPostResult) {
+				info, err := os.Stat(s.pcapfile)
+				s.NoError(err)
+				s.Len(s.payloads, 3)
+				status := s.payloads[1].(*api.PacketPostStatus)
+				s.Equal(status.Type, "PacketPostStatus")
+				s.Equal(status.PacketSize, info.Size())
+				s.Equal(status.PacketReadSize, info.Size())
+			},
+		},
+	}
+	for _, c := range cases {
+		c.Run(t)
+	}
 }
 
-func (s *PacketPostSuite) TestStatus() {
-	info, err := os.Stat(s.pcapfile)
-	s.NoError(err)
-	s.Len(s.payloads, 3)
-	status := s.payloads[1].(*api.PacketPostStatus)
-	s.Equal(status.Type, "PacketPostStatus")
-	s.Equal(status.PacketSize, info.Size())
-	s.Equal(status.PacketReadSize, info.Size())
+func (p packetPostTest) Run(t *testing.T) {
+	if p.ZeekExec == "" {
+		p.ZeekExec = zeekpath
```

Comment:
Is the use case for this to have "zeek" binaries that return specific failures?

Reply:
Yes.